### PR TITLE
fix(@angular/cli): verify Angular/TS version combos

### PR DESCRIPTION
**Pull request by filipesilva**

### CODE REVIEW

**Comment by filipesilva** [comment]
@hansl can you verify the version combos are correct?

File: packages/@ngtools/webpack/src/index.ts [hunk 1]
```diff
@@ -14,20 +16,33 @@ try {
...
-          && (   version.minor == '4'
-              || version.minor == '3' && version.patch == '1')))) {
+const versionCombos = [
+  { angular: '>=2.3.1 <3.0.0', typescript: '>=2.0.2 <2.3.0' },
+  // @chuckj says that TS 2.4 is not recommended for Angular 4, but some projects use it and
+  // failing when using it would break projects.
+  { angular: '>=4.0.0 <5.0.0', typescript: '>=2.1.0 <2.6.0' },
+  // True minimum TS for Angular 5 is 2.4.0, but 2.3.2 works with some sourcemap problems and
+  // a lot of projects are not TS 2.4 ready.
+  { angular: '>=5.0.0 <6.0.0', typescript: '>=2.3.2` <2.6.0' }
```
Comment (hansl) [inline packages/@ngtools/webpack/src/index.ts]: Seems to have a quote in there.

Comment (filipesilva) [reply packages/@ngtools/webpack/src/index.ts]: Fixed

**Comment by StephenFluin** [comment]
Is there any reason we can't rely on Angular's package.json to verify typescript versions?

**Comment by filipesilva** [comment]
@StephenFluin the `package.json` for `@angular/core` contains only these:
```
  "peerDependencies": {
    "rxjs": "^5.0.1",
    "zone.js": "^0.8.4"
  },
  "dependencies": {
    "tslib": "^1.7.1"
  },
```

Angular packages do not have a dependency on TypeScript.

**Comment by StephenFluin** [comment]
Can we make this a mutable warning instead of an error?

**Comment by filipesilva** [comment]
That's a good idea. Yes, we can.

File: packages/@angular/cli/upgrade/version.ts [hunk 2]
```diff
@@ -143,6 +143,51 @@ export class Version {
...
+      { compiler: '>=2.3.1 <3.0.0', typescript: '>=2.0.2 <2.3.0' },
+      { compiler: '>=4.0.0 <5.0.0', typescript: '>=2.1.0 <2.4.0' },
+      { compiler: '>=5.0.0 <6.0.0', typescript: '>=2.4.0 <2.6.0' }
+    ];
+
+    const currentCombo = versionCombos.find((combo) => satisfies(compilerVersion, combo.compiler));
+
+    if (currentCombo && !satisfies(tsVersion, currentCombo.typescript)) {
+      // First line of warning looks weird being split in two, disable tslint for it.
+      // tslint:disable:max-line-length
```
Comment (hansl) [inline packages/@angular/cli/upgrade/version.ts]: I don't think you have to disable max-length. The first line can be split in two (and be easier to read).

**Pull request merged**